### PR TITLE
Fix CI flakyness

### PR DESCRIPTION
**Pull request by marvinhagemeister**
This PR fixes all the weird test failures on my machine. There were several issues with our setup:

- One test ran before all scripts are loaded
- Code coverage always included external resources like google analytics
- Code coverage sometimes contains duplicate coverage information for the same resource (not sure why)
- We didn't wait for the server to boot up completely before starting our test suite (one **must** set the `port` option for `jest-puppeteer`. Only found that buried in a GitHub issue)
- Upgrade `puppeteer` + `jest-puppeteer` (just to be sure)
- We didn't caught errors when any of the requests to the GitHub API failed

### CODE REVIEW

File: test/browser/homepage.test.js [hunk 1]
```diff
@@ -2,6 +2,17 @@ jest.setTimeout(10000);
 
 const URL = 'http://localhost:4444';
 
+/**
+ * Filters out external resources and duplicate coverage information
+ */
+function getUniqueResources(arr) {
```
Comment (developit) [inline test/browser/homepage.test.js]: wow, good to know. I'm guessing this is caused by preloads? yikes.

File: test/jest-puppeteer.config.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ module.exports = {
 		devtools: process.env.HEADLESS === 'false'
 	},
 	server: {
-		command: 'PORT=4444 npm run server'
+		command: 'PORT=4444 npm run server',
+		port: 4444
```
Comment (developit) [inline test/jest-puppeteer.config.js]: heh - I had this locally and assumed it wasn't doing anything. My bad!

Comment (marvinhagemeister) [reply test/jest-puppeteer.config.js]: No worries, I just which the `jest-puppeteer` docs would mention that this is required :man_shrugging:

File: test/browser/homepage.test.js (hunk 1 above)
Comment (marvinhagemeister) [reply test/browser/homepage.test.js]: Yeah not sure why they are accessed twice. It's something we should keep an eye on.

**Pull request merged**